nuxeo-services/nuxeo-permissions/src/main/java/org/nuxeo/ecm/permissions/ACLJsonEnricher.java

```diff
@@ -122,33 +125,42 @@ public void write(JsonGenerator jg, DocumentModel document) throws IOException {
         for (ACL acl : item.getACLs()) {
             jg.writeStartObject();
             jg.writeStringField("name", acl.getName());
-            jg.writeArrayFieldStart("aces");
-            for (ACE ace : acl.getACEs()) {
-                jg.writeStartObject();
-                jg.writeStringField("id", ace.getId());
-                String username = ace.getUsername();
-                writePrincipalOrGroup(USERNAME_PROPERTY, username, jg);
-                jg.writeBooleanField("externalUser", NuxeoPrincipal.isTransientUsername(username));
-                jg.writeStringField("permission", ace.getPermission());
-                jg.writeBooleanField("granted", ace.isGranted());
-                writePrincipalOrGroup(CREATOR_PROPERTY, ace.getCreator(), jg);
-                DateTimeFormatter dateTimeFormatter = ISODateTimeFormat.dateTime();
-                jg.writeStringField("begin",
-                        ace.getBegin() != null ? dateTimeFormatter.print(new DateTime(ace.getBegin())) : null);
-                jg.writeStringField("end", ace.getEnd() != null ? dateTimeFormatter.print(new DateTime(ace.getEnd()))
-                        : null);
-                jg.writeStringField("status", ace.getStatus().toString().toLowerCase());
-
-                if (ctx.getFetched(NAME).contains(EXTENDED_ACLS_PROPERTY)) {
-                    Map<String, Serializable> m = computeAdditionalFields(document, acl.getName(), ace.getId());
-                    for (Map.Entry<String, Serializable> entry : m.entrySet()) {
-                        jg.writeObjectField(entry.getKey(), entry.getValue());
-                    }
-                }
+            writeACEsField(jg, "aces", acl, document);
+
+            ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
+            if (configurationService.isBooleanPropertyTrue(COMPATIBILITY_CONFIGURATION_PARAM)) {
```

Comment:
Can't this be looked up only once in the constructor? Or is it useless because the `ACLJsonEnricher` singleton is constructed too early?